### PR TITLE
update path parsing to support windows artifact download 

### PR DESCRIPTION
**Pull request by MartinRudzki**
This compliments PR(https://github.com/buildkite/agent/pull/1013) for windows upload, both need to be merged for artifactory to work properly.

### CODE REVIEW

File: agent/arty_downloader.go [hunk 1]
```diff
@@ -78,7 +79,7 @@ func (d ArtifactoryDownloader) Start() error {
 
 func (d ArtifactoryDownloader) RepositoryFileLocation() string {
 	if d.RepositoryPath() != "" {
-		return strings.TrimSuffix(d.RepositoryPath(), "/") + "/" + strings.TrimPrefix(d.conf.Path, "/")
+		return strings.TrimSuffix(d.RepositoryPath(), "/") + "/" + strings.TrimPrefix(filepath.ToSlash(d.conf.Path), "/")
```
Comment (lox) [inline agent/arty_downloader.go]: Can we use `path.Join` here too?

**Comment by lox** [comment]
Looks like a rogue .DS_Store is included here, is it possible to remove it?

File: agent/arty_downloader.go (hunk 1 above)
Comment (MartinRudzki) [reply agent/arty_downloader.go]: Sure will get it out soon!

**Comment by lox** [comment]
Thanks very much @MartinRudzki!

**Pull request merged**